gprofiler/utils/node.py

```diff
@@ -0,0 +1,223 @@
+from functools import lru_cache
+import shutil
+import traceback
+from typing import Callable
+from gprofiler.metadata import get_exe_version
+from gprofiler.log import get_logger_adapter
+from granulate_utils.linux.ns import get_proc_root_path, is_same_ns, NsType, run_in_ns
+from granulate_utils.linux.process import is_musl
+
+
+from . import add_permission_dir, pgrep_maps, resource_path, pgrep_exe
+
+import json
+import requests
+import os
+import signal
+import psutil
+import stat
+
+from retry import retry
```

Comment:
This part is from your previous PR https://github.com/Granulate/gprofiler/pull/438, so this PR supersedes it completely? If so, you can close that other PR.

Reply:
Actually, since I have already began reviewing it, maybe it's best if we complete it and then base this PR on top of it? To avoid losing the comments.

Reply:
All changes suggested by you are in this PR, so I think we can just close previous one

Reply:
Okie

Reply:
I will follow the comments from that PR as well.